Remotion/Validation/UnitTests/RoleCollectors/AddingPropertyValidationRuleCollector1Test.cs

```diff
@@ -73,12 +77,27 @@ public void Initialization_PropertyDeclaredInSameClass ()
       Assert.That(_addingPropertyValidationRuleCollector.CollectorType, Is.EqualTo(typeof(CustomerValidationRuleCollector1)));
       Assert.That(_addingPropertyValidationRuleCollector.Validators.Any(), Is.False);
       Assert.That(_addingPropertyValidationRuleCollector.IsRemovable, Is.False);
+      Assert.That(_addingPropertyValidationRuleCollectorType.ValidatedType, Is.EqualTo(typeof(Customer)));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void Initialization_CollectorTypeDoesNotImplementIValidationRuleCollector_ThrowsArgumentException ()
     {
+      Assert.That(
+          () => new AddingPropertyValidationRuleCollector<Customer, string>(_property, _ => "", typeof(Customer)),
+          Throws.ArgumentException
+              .With.ArgumentExceptionMessageEqualTo("Parameter 'collectorType' is a 'Remotion.Validation.UnitTests.TestDomain.Customer', which cannot be assigned to "
+                                                    + "type 'Remotion.Validation.IValidationRuleCollector'.", "collectorType"));
+    }
+
+    [Test]
+    public void Create_CollectorTypeDoesNotImplementIValidationRuleCollector_ThrowsArgumentException ()
+    {
+      Assert.That(
+          () => AddingPropertyValidationRuleCollector.Create(_userNameExpression, typeof(Customer)),
+          Throws.ArgumentException
+              .With.ArgumentExceptionMessageEqualTo("Parameter 'collectorType' is a 'Remotion.Validation.UnitTests.TestDomain.Customer', which cannot be assigned to "
+                                                    + "type 'Remotion.Validation.IValidationRuleCollector'.", "collectorType"));
     }
```

Comment:
This factory method actually belongs to `AddingPropertyValidationRuleCollectorTest.cs`. We have separate test fixtures for separates classes, including generics. That's what the 1-index in the name implies. It's short for `\`1` before I knew I could actually write `\`1`, too in file names.

Reply:
It was a bit confusing at first, then I realised that there is actually only one class `AddingPropertyValidationRuleCollector1Test.cs`. Should I create a class `AddingPropertyValidationRuleCollectorTest.cs` and move the factory method to it?

Reply:
yes, please.